django/forms/templates/django/forms/div.html

```diff
@@ -0,0 +1,26 @@
+{{ errors }}
+{% if errors and not fields %}
+  <div>{% for field in hidden_fields %}{{ field }}{% endfor %}</div>
+{% endif %}
+{% for field, errors in fields %}
+  <div{% with classes=field.css_classes %}{% if classes %} class="{{ classes }}"{% endif %}{% endwith %}>
+    {% if field.widget_type == 'radioselect' or field.widget_type == 'checkboxselectmultiple' %}
```

Comment:
This line here is currently dependant upon the widget name. I wonder if there's a way of generalising this.

Reply:
Yes... 🤔 So the answer to my question on #15317 is that we need the `use_fieldset` flag _outside_ of the `field` template. (I was looking at e.g. `multiwidget.html` &co, and how they include the input template, but the usage here wraps that, rather than being inside it.)

Reply:
I see you saw my comment. 🙂

I made this as I am working on a fuller response to your question from earlier in the week. (I wasn't sure how I can point to this line without there being a comment here 🤔).

I aim to finish writing it up my thoughts over the next couple of days.

It's expanded into something a little more general to assess the bigger picture. So in part it starts to address the question of 'what is left in crispy-forms that could be suited in core Django'.

Reply:
>  ... the `field` template. ...


This is one core elements in what I'm writing.

Crispy-forms has a core concept of a field being a  "renderable" in `field.html`. Django differs in that currently only forms and widgets have templates (calling str on a `BoundField` gives the widget rather than a whole field).

Reply:
Cool. Looking forward to it!